test/acceptance/tests.py

```diff
@@ -762,25 +783,28 @@ def verify_submission_has_peer_grade(self, learner, max_attempts=5):
             max_attempts: the maximum number of times an additional peer grading should be done
         """
         count = 0
-        while self.grade_page.wait_for_page().score is None and count < (max_attempts + 1):
+        peer_page = AssessmentPage("peer-assessment", self.browser, self.problem_loc)
+        while not peer_page.is_complete and count < (max_attempts + 1):
             count += 1
             self.do_submission_training_self_assessment("extra_{}@looping.com".format(count), None)
             self.do_peer_assessment(options=self.PEER_ASSESSMENT)
             self.login_user(learner)
             self.grade_page.visit()
 
-        self.assertIsNotNone(
-            self.grade_page.wait_for_page().score,
+        self.assertTrue(
+            peer_page.is_complete,
             "Learner still not graded after {} additional attempts".format(max_attempts)
         )
 
 
-class FullWorkflowTest(OpenAssessmentTest, FullWorkflowMixin):
+class FullWorkflowBaseTest(OpenAssessmentTest, FullWorkflowMixin):
     """
-    Tests of complete workflows, combining multiple required steps together.
+    Base class for common functionality in full workflow tests.
     """
-    def setUp(self):
-        super(FullWorkflowTest, self).setUp('full_workflow', staff=True)
+
+    # Note: this will cause a KeyError if not called from a child class
+    def setUp(self, problem_loc_key="override_plz"):
```

Comment:
Can we not pass problem_type like in OpenAssessmentTest (and why do we need to provide a default value)?